### PR TITLE
Update Docker Engine licensing info

### PR DESCRIPTION
**Pull request by usha-mandya**
Signed-off-by: Usha Mandya <usha.mandya@docker.com>

Update Docker Engine licensing info.

### CODE REVIEW

**Comment by thaJeztah** [comment]
IIUC, this is for the binaries, in which case there might be some confusion, because the rpm/deb packages of the docker cli, docker engine, and containerd that we build list "ASL 2" as license; https://github.com/docker/docker-ce-packaging/blob/v19.03.13/rpm/SPECS/docker-ce-cli.spec#L9, https://github.com/docker/containerd-packaging/blob/ba7f4c500e348f7719a6026fe4114d2d3c58f5d9/debian/copyright#L6-L15

```
License: ASL 2.0
```

(note that the license itself is appears to not be included in the static packages (`.tar.gz`) and `.deb` packages of docker-ce and docker-ce-cli currently, which is probably something that need fixing as well.

There's also a `Product License` field in the output of `docker info`, which was used for Docker Enterprise, and for Docker CE currently shows;

```
Product License: Community Engine
```


@lisaberry and perhaps @justincormack may know what needs changing in our packaging, and if those changes have to be done before the updates in our documentation, or if those can be made at a later stage.

**Comment by usha-mandya** [comment]
@lisaberry @justincormack Could you PTAL?

**Comment by justincormack** [comment]
I think its fine to update the docs first.

**Comment by AkihiroSuda** [comment]
The `Docker Engine is an open source containerization technology for building and containerizing your applications` sentence in the first paragraph would need to be changed

**Review by lisaberry — APPROVED** [review]

**Comment by thaJeztah** [comment]
@chris-crone ptal

File: engine/install/index.md [hunk 1]
```diff
@@ -146,9 +146,8 @@ Patch releases are always backward compatible with its year-month version.
 
 ### Licensing
 
-Docker is licensed under the Apache License, Version 2.0. See
-[LICENSE](https://github.com/moby/moby/blob/master/LICENSE) for the full
-license text.
+Docker Engine is licensed under the Docker Software End User License Agreement. See the
```
Comment (AkihiroSuda) [inline engine/install/index.md:149]: To minimize confusion in the community, could you clarify that the source distribution of github.com/docker/{docker,cli,...} remains free under the terms of Apache License 2.0?

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: engine/install/index.md (hunk 1 above)
Comment (AkihiroSuda) [reply engine/install/index.md:149]: Also curious, which version of Docker are covered by the Docker EULA?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
(I think discussion is still ongoing on this; let me "ask changes" to prevent merging before that completes)

File: engine/install/index.md (hunk 1 above)
Comment (justincormack) [reply engine/install/index.md:149]: Hi, we are not changing the licenses of the sources at all. This PR is not worded in any way that is agreed or intended, so closing.

**Comment by justincormack** [comment]
Please don't open this sort of change without it being agreed up front.

**Pull request closed without merging**